externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -70,43 +70,40 @@ class ExecuteStatement(
       addOperationListener()
       result = spark.sql(statement)
 
-      iter =
-        if (incrementalCollect) {
-          info("Execute in incremental collect mode")
+      val resultMaxRows = spark.conf.getOption(OPERATION_RESULT_MAX_ROWS.key).map(_.toInt)
+        .getOrElse(session.sessionManager.getConf.get(OPERATION_RESULT_MAX_ROWS))
+      iter = if (incrementalCollect) {
+        if (resultMaxRows > 0) {
+          logger.warn(s"Ignore ${OPERATION_RESULT_MAX_ROWS.key} on incremental collect mode.")
+        }
+        val internalIterator = if (arrowEnabled) {
+          SparkDatasetHelper.toArrowBatchRdd(convertComplexType(result)).toLocalIterator
+        } else {
+          result.toLocalIterator().asScala
+        }
+        new IterableFetchIterator[Any](new Iterable[Any] {
+          override def iterator: Iterator[Any] = internalIterator
```

Comment:
It changes the behavior.

before:
```
override def iterator: Iterator[Row] = result.toLocalIterator().asScala
```

now:

```
val internalIterator = result.toLocalIterator().asScala
override def iterator: Iterator[Any] = internalIterator
```

Reply:
this pr is too complex and  is more possible to hide bug

Reply:
I see your point, that's why ut `test fetch orientation with incremental collect mode` failed.

Reply:
changed `val internalIterator` to def `internalIterator()` to resolve the issue